src/main/java/org/carlmontrobotics/robotcode2023/subsystems/Arm.java

```diff
@@ -0,0 +1,369 @@
+    }
+
+    public double getArmClampedGoal(double goal) {
+        return MathUtil.clamp(MathUtil.inputModulus(goal, ARM_DISCONTINUITY_RAD, ARM_DISCONTINUITY_RAD + 2 * Math.PI), ARM_LOWER_LIMIT_RAD, ARM_UPPER_LIMIT_RAD);
+    }
+    
+    public double getWristClampedGoal(double goal) {
+        return MathUtil.clamp(MathUtil.inputModulus(goal, WRIST_DISCONTINUITY_RAD, WRIST_DISCONTINUITY_RAD + 2 * Math.PI), WRIST_LOWER_LIMIT_RAD, WRIST_UPPER_LIMIT_RAD);
+    }
+
+    // distance from center of mass relative to joint holding arm
+    public Translation2d getCoM() {
+        Translation2d comOfArm = new Translation2d(COM_ARM_LENGTH_METERS, Rotation2d.fromRadians(getArmPos()))
+                .times(ARM_MASS_KG);
+        Translation2d comOfRoller = new Translation2d(ARM_LENGTH_METERS, Rotation2d.fromRadians(getArmPos()))
+                .plus(new Translation2d(COM_ROLLER_LENGTH_METERS,
+                        Rotation2d.fromRadians(getWristPosRelativeToGround())))
+                .times(ROLLER_MASS_KG);
+        return comOfArm.plus(comOfRoller).div(ARM_MASS_KG + ROLLER_MASS_KG);
+    }
+
+    public double maxHoldingTorqueNM() {
+        return (ARM_MASS_KG + ROLLER_MASS_KG) * g * getCoM().getNorm();
+    }
+
+    public double getKg() {
+        return V_PER_NM * maxHoldingTorqueNM();
+    }
+
+    public double getArmPos() {
+        return MathUtil.inputModulus(armEncoder.getPosition(), ARM_DISCONTINUITY_RAD,
+                ARM_DISCONTINUITY_RAD + 2 * Math.PI);
+    }
+
+    public double getWristPos() {
+        return MathUtil.inputModulus(wristEncoder.getPosition(), WRIST_DISCONTINUITY_RAD,
+                WRIST_DISCONTINUITY_RAD + 2 * Math.PI);
+    }
+
+    // Unbounded wrist position relative to ground
+    public double getWristPosRelativeToGround() {
+        return getArmPos() + wristEncoder.getPosition();
+    }
+
+    public void setArmTarget(double target) {
+        goalPosRad[ARM] = getArmClampedGoal(target);
+    }
+
+    public void setWristTarget(double target) {
+        goalPosRad[WRIST] = getWristClampedGoal(target);
+    }
+
+    public void setArmWristTarget(double targetArm, double targetWrist) {
+        setArmTarget(targetArm);
+        setWristTarget(targetWrist);
+    }
+
+    public void toggleObjectType() {
+        object++;
+        object %= 2;
+    }
+
+    public void toggleFront() {
+        isFront = !isFront;
+    }
+
+    public double getArmGoal(GoalPos[] pos) {
+        if (isFront) {
+            return pos[object].armPos;
+        } else {
+            return Rotation2d.fromRadians(-Math.PI).minus(Rotation2d.fromRadians(pos[object].armPos)).plus(Rotation2d.fromDegrees(10)).getRadians();
+        }
+    }
+
+    public double getWristGoal(GoalPos[] pos) {
+        if (isFront) {
+            return pos[object].wristPos;
+        } else {
+            return Rotation2d.fromRadians(pos[object].wristPos).unaryMinus().plus(Rotation2d.fromDegrees(10)).getRadians();
```

Comment:
Positions on the opposite side can be unrelated to positions due to wrist asymmetry.